src/bin/pg_upgrade/pg_upgrade.c

```diff
@@ -558,6 +569,41 @@ copy_xact_xlog_xid(void)
 	check_ok();
 }
 
+static bool
+has_babelfish_db(ClusterInfo *cluster)
+{
+	DbInfo *db;
+	const char *db_name;
+	PGconn *conn;
+	int ntups;
+
+	for (int dbnum = 0; dbnum < cluster->dbarr.ndbs; dbnum++)
+	{
+		PGresult *res;
+
+		db = &cluster->dbarr.dbs[dbnum];
+		db_name = db->db_name;
+		prep_status("Checking %s has a Babelfish Database\n", db_name);
```

Comment:
I think we can remove this as well

Reply:
Let me move this out of the for-loop so that it only prints once.